src/core/tests/float8_e8m0.cpp

```diff
@@ -0,0 +1,378 @@
+}
+
+TEST(F8E8M0Test, f8e8m0_sig_nan) {
+    const auto val = std::numeric_limits<ov::float8_e8m0>::has_signaling_NaN;
+    EXPECT_FALSE(val);
+}
+
+TEST(F8E8M0Test, f16_quiet_nan) {
+    const auto f8 = ov::float8_e8m0(std::numeric_limits<ov::float16>::quiet_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b11111111);
+    EXPECT_EQ(std::numeric_limits<ov::float8_e8m0>::quiet_NaN().to_bits(), 0b11111111);
+}
+
+TEST(F8E8M0Test, f16_sig_nan) {
+    const auto f8 = ov::float8_e8m0(std::numeric_limits<ov::float16>::signaling_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b11111111);
+}
+
+TEST(F8E8M0Test, f32_quiet_nan) {
+    const auto f8 = ov::float8_e8m0(std::numeric_limits<float>::quiet_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b11111111);
+}
+
+TEST(F8E8M0Test, f32_sig_nan) {
+    const auto f8 = ov::float8_e8m0(std::numeric_limits<float>::signaling_NaN());
+
+    EXPECT_TRUE(std::isnan(f8));
+    EXPECT_EQ(f8.to_bits(), 0b11111111);
+}
+
+TEST(F8E8M0Test, f8e8m0_min_normalized) {
+    const auto f8 = ov::float8_e8m0(5.877471754111437e-39f);
+
+    EXPECT_EQ(f8.to_bits(), 0b00000000);
+    EXPECT_EQ(f8.to_bits(), std::numeric_limits<ov::float8_e8m0>::min().to_bits());
+}
+
+TEST(F8E8M0Test, f8e8m0_max_normalized) {
+    const auto f8 = ov::float8_e8m0(1.7014118346046923e+38f);
+
+    EXPECT_EQ(f8.to_bits(), 0b11111110);
+    EXPECT_EQ(f8.to_bits(), std::numeric_limits<ov::float8_e8m0>::max().to_bits());
+}
+
+TEST(F8E8M0Test, f8e8m0_lowest_normalized) {
+    const auto f8 = ov::float8_e8m0(5.877471754111437e-39f);
+
+    EXPECT_EQ(f8.to_bits(), 0b00000000);
+    EXPECT_EQ(f8.to_bits(), std::numeric_limits<ov::float8_e8m0>::lowest().to_bits());
+}
+
+TEST(F8E8M0Test, f32_ge_f8_round_to_max) {
+    const auto f8 = ov::float8_e8m0(3.402823466385288598117042E+38);
+
+    EXPECT_EQ(f8.to_bits(), 0b11111110);
+}
+
+TEST(F8E8M0Test, f8e8m0_num_limits_exp) {
+    const auto min_exp = std::numeric_limits<ov::float8_e8m0>::min_exponent;
+    const auto min_exp10 = std::numeric_limits<ov::float8_e8m0>::min_exponent10;
+    const auto max_exp = std::numeric_limits<ov::float8_e8m0>::max_exponent;
+    const auto max_exp10 = std::numeric_limits<ov::float8_e8m0>::max_exponent10;
+
+    EXPECT_EQ(min_exp, -126);
+    EXPECT_EQ(min_exp10, -38);
+    EXPECT_EQ(max_exp, 128);
+    EXPECT_EQ(max_exp10, 38);
+}
+
+template <class TContainer>
+std::vector<std::tuple<uint8_t, typename TContainer::value_type>> enumerate(const TContainer& values) {
+    std::vector<std::tuple<uint8_t, typename TContainer::value_type>> enum_values;
+    uint8_t i = 0;
+    for (const auto& v : values) {
+        enum_values.emplace_back(i++, v);
+    }
+    return enum_values;
+}
+
+constexpr auto f32_nan = std::numeric_limits<float>::quiet_NaN();
+constexpr float _2_power_127 = 1.7014118346046923e+38f;
+constexpr float _2_power_negative_127 = 5.877471754111437e-39f;
```

Comment:
```suggestion
constexpr float two_power_127 = 1.7014118346046923e+38f;
constexpr float two_power_negative_127 = 5.877471754111437e-39f;
```
To avoid start names from `_`

Reply:
done